### ISSUE TITLE
Class 'SqlParser\Lexer' not found (OS X)

### ISSUE THREAD

**Issue by lem9** [phpmyadmin/phpmyadmin#11434]
QA_4_5, After a failed login:
````
PHP Fatal error:  Class 'SqlParser\Lexer' not found in libraries/Util.class.php on line 632
````

**Comment by lem9** [comment]
Sorry but the problem is not fixed for me, it just moved to line 634.

**Comment by madhuracj** [comment]
@lem9 
I could create the problem with config authentication with a wrong password and this was fixed with the about commit.
Which authentication method did you use? Can you paste the full stack trace here.

**Comment by udan11** [comment]
I tried using `cookie` and `http` authentications and I couldn't reproduce the bug.

**Comment by lem9** [comment]
MySQL 5.7.8, cookie auth.
````
PHP Fatal error:  Class 'SqlParser\\Lexer' not found in libraries/Util.class.php on line 634, referer: http://localhost/phpmyadmin/index.php

PHP Stack trace:, referer: http://localhost/phpmyadmin/index.php
PHP   1. {main}() index.php:0, referer: http://localhost/phpmyadmin/index.php
PHP   2. require_once() index.php:12, referer: http://localhost/phpmyadmin/index.php
PHP   3. PMA_DatabaseInterface->connect() /libraries/common.inc.php:992, referer: http://localhost/phpmyadmin/index.php
PHP   4. PMA_DatabaseInterface->postConnect() /libraries/DatabaseInterface.class.php:2752, referer: http://localhost/phpmyadmin/index.php
PHP   5. PMA_DatabaseInterface->query() /libraries/DatabaseInterface.class.php:1844, referer: http://localhost/phpmyadmin/index.php
PHP   6.PMA_Util::mysqlDie() /libraries/DatabaseInterface.class.php:95, referer: http://localhost/phpmyadmin/index.php
````

**Comment by madhuracj** [comment]
This is going to be a bit of chicken and egg problem. 
Lexer is called due to an error while trying to detect whether it's Drizzle where as knowing whether it's Drizzle is required to initialize Parser and Lexer.

**Comment by udan11** [comment]
This is weird because this problem didn't exist before the parser was replaced and I tried not to move `include` statements so I would not mess the dependencies.

**Comment by lem9** [comment]
I can only reproduce this under MySQL 5.7.

**Comment by lem9** [comment]
This is triggered by SHOW SESSION VARIABLES not being allowed for unprivileged users, but I'll try to fix the parser error while the SHOW error is not fixed yet :)

**Comment by lem9** [comment]
See https://github.com/phpmyadmin/phpmyadmin/issues/11442

**Comment by y2kbadbug** [comment]
I'm on master and this bug is affecting me.  Our configuration is cookie auth, we have multiple servers configured.  Only some servers produce this error when logging in.

**Comment by kpfl** [comment]
It's also triggered here. The query which caueses this bug is:
SET collation_connection = 'utf8mb4_unicode_ci';

The destination server has no such collation only the source server.

Source server is MariaDB, destination server is MySQL.

But the same works for other MySQL Servers.

Why did the autoloader not find that class?

**Comment by kpfl** [comment]
If I write in line 630

require_once __DIR__ . '/sql-parser/autoload.php';

I see the error from MySQL about the unknown collation.

**Comment by svivian** [comment]
Which version is this fixed in? I'm using PMA 4.5.3.1 and still getting the problem. It happens for me when (a) my session has timed out and (b) I restart my browser with the PMA tab open. The error is same as above:

> Fatal error: Class 'SqlParser\Lexer' not found in /.../phpmyadmin/libraries/Util.class.php on line 634

Using PHP 5.5.30, MySQL 5.6.13.

The URL it's trying to load is like

    http://localhost/phpmyadmin/sql.php?db=mydatabase&table=users&pos=0&sql_query=SELECT+%2A+FROM+%60mydatabase%60.%60users%60+WHERE+%60id%60+%3D+1234&token=692a00e174577c323398eeef66168270

**Comment by lem9** [comment]
@svivian This is supposed to have been fixed in 4.5.0, as the "milestone" above is stating. Is your server OS X ?

**Comment by svivian** [comment]
Yes OSX, using the GitHub repo (stable branch).

**Comment by cmrunton** [comment]
I'm also seeing this error when trying to connect on an initial login, from an AWS EC2 instance to an AWS RDS instance. Using Cookie auth, single server configured. 

Versions:
RDS MySQL -> 5.6.23
EC2 PHP -> 7.0.2-1
EC2 phpMyAdmin -> 4.5.3.1 - English

**Stacktrace from nginx error log** 

    2016/01/14 19:53:18 [error] 1269#1269: *99 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Class 'SqlParser\Lexer' not found in /usr/share/nginx/html/logthecrux/admin/phpmyadmin-4.5.3.1/libraries/Util.class.php:634
    Stack trace:
    #0 /usr/share/nginx/html/logthecrux/admin/phpmyadmin-4.5.3.1/libraries/DatabaseInterface.class.php(95): PMA_Util::mysqlDie('#2006 - MySQL s...', 'SELECT * FROM i...')
    #1 /usr/share/nginx/html/logthecrux/admin/phpmyadmin-4.5.3.1/libraries/mysql_charsets.inc.php(23): PMA_DatabaseInterface->query('SELECT * FROM i...')
    #2 /usr/share/nginx/html/logthecrux/admin/phpmyadmin-4.5.3.1/libraries/common.inc.php(1063): include_once('/usr/share/ngin...')
    #3 /usr/share/nginx/html/logthecrux/admin/phpmyadmin-4.5.3.1/index.php(12): require_once('/usr/share/ngin...')
    #4 {main}
      thrown in /usr/share/nginx/html/logthecrux/admin/phpmyadmin-4.5.3.1/libraries/Util.class.php on line 634" while reading response header from upstream, client: ***.***.***.***, server: pma.logthecrux.com, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.0-fpm.sock:", host: "pma.logthecrux.com"

**Comment by svivian** [comment]
Another thing, not sure if related - when I run a query and then click the "Refresh" link (on the query box after Edit inline / Edit / ...) the URL changes to `/import.php` for some reason. The last couple of times I had the SqlParser error the URL was `/import.php`.

**Comment by madhuracj** [comment]
While I am able to reproduce the error on 4.5 branch, I am unable to reproduce this on 4.6 branch. @svivian @cmrunton Are you in a position to try out phpMyAdmin 4.6.0-alpha1 (https://www.phpmyadmin.net/downloads/) and tell us whether the issues in gone?

**Comment by svivian** [comment]
@madhuracj How are you reproducing the issue? It's very sporadic for me, I can't reliably reproduce the error.

**Comment by madhuracj** [comment]
@svivian I recreated by manually introducing a syntax error in the SQL query that queries for character sets (https://github.com/phpmyadmin/phpmyadmin/blob/QA_4_5/libraries/mysql_charsets.inc.php#L22).
This is included (https://github.com/phpmyadmin/phpmyadmin/blob/QA_4_5/libraries/mysql_charsets.inc.php#L22) prior to the SQL Parser (https://github.com/phpmyadmin/phpmyadmin/blob/QA_4_5/libraries/common.inc.php#L1094)

**Comment by timri** [comment]
I can reproduce this error on Amazon EC2 (accessing RDS):
When I enable ssl by including:
```
$cfg['Servers'][$i]['ssl'] = true;
$cfg['Servers'][$i]['ssl_key'] = null;
$cfg['Servers'][$i]['ssl_cert'] = null;
$cfg['Servers'][$i]['ssl_ca'] = '/etc/ssl/rds-combined-ca-bundle.pem';
$cfg['Servers'][$i]['ssl_ca_path'] = null;
$cfg['Servers'][$i]['ssl_ciphers'] = null;
```
into config.inc.php I get an "internal server error" in the browser and the error-message about SQLParser\Lexer in the error-log (with phpmyadmin-4.5.4.1) directly after login.

Trying phpmyadmin-4.6.0-alpha, I get:
> Fehler
> SQL-Befehl:
> SELECT * FROM information_schema.CHARACTER_SETS
> MySQL meldet:
> #2006 - MySQL server has gone away

In both versions, I am able to login and use it after I change the config to
```
$cfg['Servers'][$i]['ssl'] = false;
```

**Comment by cmrunton** [comment]
@madhuracj I was able to install the 4.6.0-RC2 today. I'm now getting a 500 Internal Server Error when I try to access the login page, and the following error thrown in my server logs:

    2016/03/02 23:05:07 [error] 6479#6479: *1265 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Call to undefined function __() in /usr/share/nginx/html/logthecrux/admin/phpMyAdmin-4.6.0-rc2-english/libraries/sanitizing.lib.php:135
    Stack trace:
    #0 /usr/share/nginx/html/logthecrux/admin/phpMyAdmin-4.6.0-rc2-english/libraries/Message.php(586): PMA_sanitize('Undefined index...', false, true)
    #1 /usr/share/nginx/html/logthecrux/admin/phpMyAdmin-4.6.0-rc2-english/libraries/Message.php(650): PMA\libraries\Message::decodeBB('Undefined index...')
    #2 /usr/share/nginx/html/logthecrux/admin/phpMyAdmin-4.6.0-rc2-english/libraries/Error.php(167): PMA\libraries\Message->getMessage()
    #3 /usr/share/nginx/html/logthecrux/admin/phpMyAdmin-4.6.0-rc2-english/libraries/ErrorHandler.php(157): PMA\libraries\Error->getHash()
    #4 /usr/share/nginx/html/logthecrux/admin/phpMyAdmin-4.6.0-rc2-english/libraries/ErrorHandler.php(121): PMA\libraries\ErrorHandler->addError('Undefined index...', 8, '/usr/share/ngin...', 285, true)
    #5 /usr/share/nginx/html/logthecrux/admin/phpMyAdmin-4.6" while reading response header from upstream, client: ***.***.***.***, server: pma.logthecrux.com, request: "GET /index.php HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.0-fpm.sock:", host: "pma.logthecrux.com"

Let me know if you have any insight, and I'd be happy to test more.

**Comment by madhuracj** [comment]
@cmrunton Can you try deleting cookies for the site and clearing your browser cache.

**Comment by cmrunton** [comment]
@madhuracj No change when I clear caches and delete cookies. Same errors from the server.

**Comment by svivian** [comment]
I'm also using 4.6-rc2. Now getting a different error, in the same "conditions" as the previous bug. (For reference, the condition is when the session times out and PMA auto logs out.) The error is:

```
SQL query:

INSERT INTO `phpmyadmin`.`pma__history`
              (`username`,
                `db`,
                `table`,
                `timevalue`,
                `sqlquery`)
        VALUES
              ('root',
               '[mydatabase]',
               '[mytable]',
               NOW(),
               '[last query I ran]')
```

That's PMA outputting the error, not PHP itself (I have xdebug so PHP errors show with that orange table.)

Apache 2.4.16 on OSX
PHP 5.5.30
MySQL 5.6.13
Git revision: 0fc00c0

**Comment by svivian** [comment]
Update: strangely this issue is only occurring in Google Chrome, and not in Firefox. In FF it auto-logs out just fine. So many it's a problem with the JavaScript? Or Chrome is doing something else weird when the session times out.

**Comment by nijel** [comment]
After reading this issue, I'm not really able to understand the problem here. It started as error about not finding some class and it ended up with some completely different errors. As the original issue should be resolved (nobody was able to reproduce it in 4.6.0), I'm closing this issue
